### PR TITLE
ILIKE operator

### PR DESCRIPTION
**Pull request by myrrc**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Implementation of PostgreSQL-like ILIKE operator for  https://github.com/ClickHouse/ClickHouse/issues/11710

Detailed description / Documentation draft:

Usage similar to LIKE.

### CODE REVIEW

File: src/Functions/like.cpp [hunk 1]
```diff
@@ -11,11 +11,28 @@ struct NameLike
...
+
+using FunctionLike = FunctionsStringSearch<LikeImpl, NameLike>;
+using FunctionIlike = FunctionsStringSearch<ILikeImpl, NameILike>;
 
 void registerFunctionLike(FunctionFactory & factory)
 {
     factory.registerFunction<FunctionLike>();
 }
 
+void registerFunctionIlike(FunctionFactory & factory)
```
Comment (alexey-milovidov) [inline src/Functions/like.cpp:33]: I prefer creating a separate file `ilike.cpp`
BTW, do we have NOT ILIKE operator?

**Review by alexey-milovidov — COMMENTED** [review]
Need to check how it will work with UTF-8 in case of regexp.
`'ёЁё' ILIKE 'Ё%Ё'`

File: src/Functions/like.cpp (hunk 1 above)
Comment (myrrc) [reply src/Functions/like.cpp:33]: > I prefer creating a separate file `ilike.cpp`

ILIKE and LIKE differ only in the searcher (VolnitskyCaseInsensitive vs. Volnitsky), so I don't think this is necessary (I wouldn't rather pollute the file structure).
Same is true for NOT LIKE and NOT ILIKE.

> BTW, do we have NOT ILIKE operator?

Just discussed with my colleague!
Not yet, but I can add it.

**Comment by myrrc** [comment]
> Need to check how it will work with UTF-8 in case of regexp.
> `'ёЁё' ILIKE 'Ё%Ё'`

I'll push the tests today.

File: tests/queries/0_stateless/01355_ilike.sql [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+SELECT 'ощщЁё' ILIKE '%щ%';
+SELECT 'ощЩЁё' ILIKE '%ё%';
+
+SHOW TABLES NOT ILIKE '%';
+DROP DATABASE IF EXISTS test_01355;
+CREATE DATABASE test_01355;
+USE test_01355;
+CREATE TABLE test1 (x UInt8) ENGINE = Memory;
+CREATE TABLE test2 (x UInt8) ENGINE = Memory;
+SHOW TABLES ILIKE 'tes%';
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/01355_ilike.sql:59]: But the test case does not show the difference of ILIKE...

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
(Will merge with squash)

**Comment by alexey-milovidov** [comment]
> Yandex synchronization check

Bogus result, our friends from Arcadia will investigate.

**Pull request merged**